### ISSUE TITLE
enum prefix disable not working

### ISSUE THREAD

**Issue by veeramarni** [dotansimha/graphql-code-generator#2679]
**Describe the bug**
 
When using the following options for`enum`, it generates correctly without a prefix but they are referred with the prefix at `export type IResolversTypes` and typescript compilation fails.

```
  config:
    typesPrefix: I
    enumPrefix: false
```

**To Reproduce**

1. My GraphQL schema:

```graphql
enum GroupLocation {
    FIRST
    LAST
    NEXT
    PREVIOUS
}
```

3. My `codegen.yml` config file:

```yml
generates:
  package/generated-model/generated-model.ts:
    schema:
      - servers/backend-server/src/api/root-schema.graphqls
    documents: 
      - packages-modules/extension/browser/src/graphql/**/*.gql
    config:
        noNamespaces: true
        contextType: ../apollo-context#MyContext
        typesPrefix: I
        enumPrefix: false
        namingConvention: keep
        constEnums: true
        withMutationFn: false
        withHOC: false
        withComponent: false
    plugins:
      - add: |
          /* tslint:disable */
      - typescript
      - typescript-operations
      - typescript-resolvers
      - typescript-react-apollo
```

Output

```
/** Mapping between all available schema types and the resolvers types */
export type IResolversTypes = {
GroupLocation: IGroupLocation,
.....
}

```
**Expected behavior**

No prefix when referring to Enums.

**Environment:**

- OS: 
- `@graphql-codegen/...`:  1.8.0
- NodeJS:

**Comment by KyleMcNutt** [comment]
Just ran into this bug as well, or one somewhat similar to it.

In my case, if I had a type that has a property referring to an enum, it would prefix the enum name with an `I`, even though, the enum wasn't prefix.

i.e.:

```
enum FilterOption {
    New,
    Active,
    Closed
}

input UpdateFilterOptionInput {
  newOption: FilterOption!
}
```

It would generate code like:

```
export enum FilterOption {
    New = 'New',
    Active = 'Active',
    Closed = 'Closed'
}

export type IUpdateFilterOptionInput = {
  // Compiler error IFilterOption doesn't exist.
  newOption: IFilterOption
}
```

**Comment by KyleMcNutt** [comment]
From doing a bit of digging, looks like this bug might be here:

https://github.com/dotansimha/graphql-code-generator/blob/2fe0ec94cd14bd898919b2caa6bac5c1783eb47f/packages/plugins/other/visitor-plugin-common/src/base-visitor.ts#L150

Where I believe this should take into account the `enumPrefix` config setting?

**Comment by dotansimha** [comment]
Working on a fix here: https://github.com/dotansimha/graphql-code-generator/pull/2759

**Comment by dotansimha** [comment]
@KyleMcNutt can you please explain how to reproduce your issue? I tried here: https://github.com/dotansimha/graphql-code-generator/blob/383c7122ca7da2f973bdf48284c264e7694844f9/packages/plugins/typescript/typescript/tests/typescript.spec.ts#L228 

but it seems fine...

**Comment by KyleMcNutt** [comment]
@dotansimha Yeap, looks like bumping my version fixed the issue. I bumped from `1.6.1` to `1.8.1` and this bug went away. Thanks for taking a look!

**Comment by dotansimha** [comment]
Fixed in 1.8.2.
